### ISSUE TITLE
SRD0018 false positive (and SRD0014 not triggered)

### ISSUE THREAD

**Issue by ErikEJ** [ErikEJ/SqlServer.Rules#716]
```
CREATE TABLE dbo.test (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

UPDATE dst
SET currentId = bodyItems.currentId
FROM dbo.test dst
INNER JOIN OPENJSON(@body, '$.data') WITH (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
) bodyItems
ON bodyItems.originalId = dst.originalId;
```
Fine, SRD0018 doesn't pop up.

But, it pops up for this
```
DECLARE @someTable TABLE (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

UPDATE dst
SET currentId = bodyItems.currentId
FROM dbo.test dst
INNER JOIN @someTable bodyItems
ON bodyItems.originalId = dst.originalId;
```
@niphlod FYI

**Comment by ErikEJ** [comment]
@niphlod

I do not see SRD0018 here:

```sql
CREATE TABLE dbo.test (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

DECLARE @someTable TABLE (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

UPDATE dst
SET currentId = bodyItems.currentId
FROM dbo.test dst
INNER JOIN @someTable bodyItems
ON bodyItems.originalId = dst.originalId;

```

**Comment by niphlod** [comment]
@ErikEJ , sorry for the delay, I was on holidays ^_^


something weird is going on.

If you run 

```
CREATE TABLE dbo.test (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);
```

then 
```
DECLARE @someTable TABLE (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

UPDATE dst
SET currentId = bodyItems.currentId
FROM dbo.test dst
INNER JOIN @someTable bodyItems
ON bodyItems.originalId = dst.originalId;
```

triggers both SRD0018 and SRP0014.

Drop the dbo.test table......
open a new query and add the very same create table at the top .... it doesn't trigger anything
```
CREATE TABLE dbo.test (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

DECLARE @someTable TABLE (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);

UPDATE dst
SET currentId = bodyItems.currentId
FROM dbo.test dst
INNER JOIN @someTable bodyItems
ON bodyItems.originalId = dst.originalId;
```

**Comment by ErikEJ** [comment]
@niphlod I am struggling to trigger the 2 warnings - could you provide more detailed repro steps?

**Comment by niphlod** [comment]
fresh db. I'm on the latest 1.0.962.

step 1

```
CREATE TABLE dbo.test (
	originalId  VARCHAR(50),
	currentId   VARCHAR(50)
);
```
execute, close the window.


2. open a fresh window, paste the query

<img width="900" height="581" alt="Image" src="https://github.com/user-attachments/assets/9699348f-bdbe-4573-b14f-179965a3c8da" />

both rules are triggered.
